packages/documentation/src/pages/getting-started/common-tasks/new-unit-test.mdx

```diff
@@ -0,0 +1,141 @@
+        ...
+        expect(someCondition).to.be.true;
+
+        componentUnderTest.unmount();
+    })
+  });
+});
+```
+
+- use `describe` to organize tests by application and feature
+  - two levels is adequate depending on the size of the application being tested
+- use `it` to describe the unit test
+  - use active voice
+  - describe the behavior in terms specific to the unit
+  - don't abstract the test description with business logic
+  - **Recommended**:
+    - `it('truncates the address property when it is longer than 15 characters')`
+    - `it('renders an error when props.errors contains at least one item')`
+  - **Not recommended**:
+    - `it('shortens the address when the user has a long address')`
+    - `it('shows an error when the user is not logged in')`
+  - use Enzyme's [shallow](https://airbnb.io/enzyme/docs/api/shallow.html) instead of [mount](https://airbnb.io/enzyme/docs/api/mount.html) when possible to test components
+  - always `unmount` components at the end of the test
+
+## Unit testing forms with schemaform-utils
+
+```
+import { DefinitionTester } from 'platform/testing/unit/schemaform-utils';
+
+describe('MyForm FormID', () => {
+
+const {
+    schema,
+    uiSchema,
+    arrayPath,
+  } = formConfig.chapters.myFormChapter.pages.myFormPage;
+
+  it('renders myFormPage form', () => {
+    const form = mount(
+      <DefinitionTester
+        arrayPath={arrayPath}
+        pagePerItemIndex={0}
+        definitions={formConfig.defaultDefinitions}
+        schema={schema}
+        data={initialData}
+        formData={initialData}
+        uiSchema={uiSchema}
+      />,
+    );
+
+    expect(form.find('select').length).to.equal(6); // from/to months, days; country, state
+    expect(form.find('input').length).to.equal(4); // facility name, from/to years, city
+    form.unmount();
+  });
+  ```
+
+- organize forms config tests by page
+- use the `<DefinitionTester />` to render form configs for testing
+- test at least
+  - number of each type of input is rendered (example above)
+  - any conditional display logic on the page
+  - all field level validation errors
+- use the other `schemaform-utils` for filling out form data:
+
+>- `fillData()` - Enzyme helper that fires a change event with a value for an element at the given selector
+```
+fillData(
+  form, // mounted <DefinitionTester />
+  'select#root_blah' // selector
+  'USA', // value
+);
+```
+
+>- `fillDate()` - Enzyme helper that fills in a date field with data from the given date string.
+```
+fillDate(
+  form, // mounted <DefinitionTester />
+  'select#root_blah' // selector
+  '1950-1-3', // value
+);
+```
+
+>- `selectCheckbox()` - Enzyme helper that fires a change event with a value for a checkbox at the given name
+
+## Other utilities
+
+_These utilities can be found in `platform/testing/unit/helpers.js`_
+
+- `mockFetch()` -  A function to mock the global fetch function and return the value provided in returnVal
```

Comment:
I might be missing a lot of utilities here. 

- there are two sets of utilities for doing forms input: the `platform/testing/unit/helpers` and `schemaform-utils` have several utilities with the same names 
- there are a few utilities in these files that are used once or not at all- i left these out. 
- all of the utilities have good JSDOC comments. might be useful to figure out a way to pull those into this repo and link to them in these docs 
  - might be useful if we could reference blurbs of content in markdown - e.g. we could document all of the utilities in an index but reference them by a name / markdown code that automatically inlines the jsdoc info and links to the full documentation. 
  - Good example is apple's ios docs: https://developer.apple.com/documentation/uikit/views_and_controls
  - Links to other docs have a summary of what the item is about

Reply:
Yeah, I definitely want to get some of our JSDoc comments into the doc site.